### PR TITLE
#3730 Add new pref to enable editing of all patients

### LINKED ISSUE #3730: Handle the pref to enable editing of all patients
## Is your feature request related to a problem? Please describe.

There is a new pref on desktop: `can_edit_patients_from_any_store`

## Describe the solution you'd like

When this preference is enabled, allow editing all patients.

## Implementation

TBD

## Describe alternatives you've considered

TBD

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #3730 

## Change summary

- Adds the pref to enable editing of all patients 🙃 

## Testing

- [ ] Looking up a patient from the lookup modal while having this pref

### CODE REVIEW

File: src/database/DataTypes/Name.js [hunk 1]
```diff
@@ -4,6 +4,8 @@
  */
 
 import Realm from 'realm';
+import { UIDatabase } from '../index';
+import { PREFERENCE_KEYS } from '../utilities/preferenceConstants';
```
Comment (joshxg) [inline src/database/DataTypes/Name.js:8]: This isn't ideal- adding UIDatabase import is a circular reference .. 

Alternatives:

- Put it into the `DataTableRow` (which is calling the getter on an instance of this class). Not great as it means this getter is actually not correct
- Put it into the component which calls `DataTableRow` and pass as some sort of `option` prop... not great as there is already plenty of props and the condition for enabling/disabling a row is already really crazy in `DataTableRow`, but the advantage is that this menans there's one db call

Comment (mark-prins) [reply src/database/DataTypes/Name.js:8]: would it be crazy to lookup preferences when the app starts and store them globally?
Update the state on incoming sync?

**Review by mark-prins — APPROVED** [review]
I wonder about storing prefs in state, rather than looking up in realm each time. maybe later though?

**Comment by joshxg** [comment]
🤔 

Maybe a good alternative. Also thought of saving editability against the name records themselves... (and not syncing back)

I think it would be good to do some testing with a lot of name records anyway, so we can see if it makes a difference there. Realm is really quick!

**Pull request merged**